aws-s3-transfer-manager/tests/e2e_transfer_test.rs

```diff
@@ -305,3 +304,89 @@ async fn test_upload_with_long_running_stream() {
         }
     }
 }
+
+#[tokio::test]
+async fn test_empty_object_download() {
+    let (tm, _) = test_tm().await;
+    let (bucket_name, _) = get_bucket_names();
+
+    let object_key = "pre-existing-empty";
+
+    let mut download_handle = tm
+        .download()
+        .bucket(bucket_name)
+        .key(object_key)
+        .initiate()
+        .unwrap();
+
+    let body = drain(&mut download_handle).await.unwrap();
+
+    assert_eq!(body.len(), 0);
+}
+
+async fn range_download_helper(
+    tm: &aws_s3_transfer_manager::Client,
+    bucket: &str,
+    key: &str,
+    range: &str,
+    expected_length: usize,
+) -> Result<(), aws_s3_transfer_manager::error::Error> {
+    let mut download_handle = tm
+        .download()
+        .bucket(bucket)
+        .key(key)
+        .range(range)
+        .initiate()?;
+
+    let body = drain(&mut download_handle).await?;
+
+    assert_eq!(body.len(), expected_length);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_object_download_range() {
```

Comment:
trivial/ debatable: It might be cleaner to split success/fail cases into two separate test_cases.